logstash-core/lib/logstash/pipeline.rb

```diff
@@ -379,12 +379,16 @@ def plugin(plugin_type, name, *args)
     klass = LogStash::Plugin.lookup(plugin_type, name)
 
     if plugin_type == "output"
-      LogStash::OutputDelegator.new(@logger, klass, *args)
+      LogStash::OutputDelegator.new(@logger, klass, default_output_workers, *args)
     else
       klass.new(*args)
     end
   end
 
+  def default_output_workers
+    @settings[:pipeline_workers] || @settings[:default_pipeline_workers]
+  end
```

Comment:
This I think is separate, because `default_output_workers` is semantically different from `default_pipeline_workers`, yet it must move in lock-step with the default value of the pipeline workers. Are you sure you aren't confusing the two? IMHO this is the correct place for this.